python/docs/index.rst

```diff
@@ -1,11 +1,134 @@
    :maxdepth: 2
 
-   tutorial
+   proton
+   proton.handlers
+   proton.reactor
+   proton.utils
+
+*****************************************
+About AMQP and the Qpid Proton Python API
+*****************************************
+
+.. toctree::
+   :maxdepth: 1
+
    overview
+   tutorial
+
+Key API Features
+================
+
+ * Event-driven API
+ * SSL/TLS secured communication
+ * SASL authentication
+ * Automatic reconnect and failover
+ * Seamless conversion between AMQP and Python data types
+ * AMQP 1.0
+
+Basic API Concepts
+==================
+
+The Qpid Python client API and library allows applications to send and receive AMQP messages. See :ref:`overview`
+for a more detailed explanation.
+
+Containers
+----------
+
+Messages are transferred between connected peers (or nodes) using **senders** and **receivers**. Each sender
+or receiver is established over a **connection**. Each connection is established between two unique **containers**,
+the entry point for the API. The container class :class:`proton.reactor.Container` is found in the ``proton.reactor``
+module.
+
+Senders
+-------
+
+The peer that sends messages uses a **sender** to send messages, which includes the target queue or topic which is
+to receive the messages. The sender may be found at :class:`proton.Sender`. Note that senders are most commonly
+obtained by using the convenience method :meth:`proton.reactor.Container.create_sender`.
+
+Receivers
+---------
+
+The peer that receives messages uses a **receiver** to receive messages, and includes a source queue or topic from
+which to receive messages. The receiver may be found at :class:`proton.Receiver`. Note that senders are most commonly
+obtained by using the convenience method :meth:`proton.reactor.Container.create_receiver`.
+
+Message Delivery
+----------------
+
+The process of sending a message is known as **delevery**. Each sent message has a delivry object which tracks the
+status of the message as it is sent from the sender to the receiver. This also includes actions such as settling the
+delivery, a mechanism by which each peer can verify that the message has been properly received and is either accepted
+or rejected. The delivery class may be found at :class:`proton.Delivery`. The delivery object is most commonly obtained
+when a message-related event occurs through the event object. See `Event Handlers`_ below.
+
+Event Handlers
+--------------
+
+A **handler** is a class that handles events associated with the sending and receiving of messages. This includes
+callbacks for events such as the arrival of a new message, error conditions that might arise, and the closing
+of the connection over which messages are sent. An application developer must handle some key events to
+successfully send and receive messages. When an event handler callback is called by the library, an Event object is
+passed to it which contains details of the event, and in some cases, objects associated with the event. For example,
```

Comment:
Really the only point of the Event object passed to the handlers is to get the object to which the event pertains. It also contains the event handled (but we know what that is because clearly we're already in the appropriate handler.

Reply:
So strike 'in some cases'

Reply:
I changed this to read:

> When an event handler callback is called by the library, an Event object is
> passed to it which contains an object associated with the event.